docs/source/design/iom.rst

```diff
@@ -0,0 +1,86 @@
+.. _iom:
+
+==================
+Writing Inlet oulet manager (IOM)
+==================
+
+This section essentially talks about writing your own inlet outlet manager.
+If you want to use the existing IOM subclass present in PySPH see :ref:`flow_past_cylinder`. The IOM manages all the inputs required to simulate open boundaries in PySPH. It has the following functions
+
+* Create ghost particles
+* Create Inlet outlet stepper
+* Creation of Inlet outlet equations
+* Creation of Inlet outlet updater
+
+Overview
+~~~~~~~~
+
+.. py:currentmodule:: pysph.sph.bc.inlet_outlet_manager
+
+The brief Overview of :py:class:`InletOutletManager` subclass:: 
+
+    class MyIOM(InletOutletManager):
+        def __init__(self, fluid_arrays, inletinfo, outletinfo,
+                    extraeqns=None):
+            # Create the object to manage inlet outlet boundary conditions.
+            # Most of the variables are evaluated after the scheme and particles
+            # are created.
+
+        def create_ghost(self, pa_arr, inlet=True):
+            # Creates ghosts for the given inlet/outlet particles
+            # return ghost_pa ( the ghost particle array for the pa_arr)
+
+        def update_dx(self, dx):
+            # Update the discretisation length
+
+        def add_io_properties(self, pa, scheme=None):
+            # Add properties to be used in inlet/outlet equations tp pa
+            # pass
+
+        def get_io_names(self, ghost=False):
+            # return all the names of inlets and outlets
+            
+        def get_stepper(self, scheme, integrator, **kw):
+            # Returns the steppers for inlet/outlet
+            raise NotImplementedError()
+
+        def setup_iom(self, dim, kernel):
+            # Essential data passed
```

Comment:
Not clear what this essential data is?

Reply:
Why isn't this in the `__init__`?

Reply:
It is required to pass information from the consume_user_data. Added this in the documentation.